dot/parachain/candidate-validation/worker_pool.go

```diff
@@ -82,6 +83,10 @@ const (
 	CodeHashMismatch
 	// CommitmentsHashMismatch Validation has generated different candidate commitments.
 	CommitmentsHashMismatch
+	// InvalidSessionIndex The candidate receipt contains an invalid session index.
+	InvalidSessionIndex
+	// The candidate receipt contains an invalid core index.
+	InvalidCoreIndex
 )
 
 func (ci ReasonForInvalidity) Error() string {
```

Comment:
You could add a `Clone()` method on this type and use it instead of the copy-by-assignment to put it into a `ValidationResult`.

```go
func (ci ReasonForInvalidity) Clone() &ReasonForInvalidity {
        c := ci
        return &c
}
// ...

return &ValidationResult{Invalid: InvalidCoreIndex.Clone()}, nil
```